model/DataObject.php

```diff
@@ -1744,63 +1756,90 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
 	}
 
 	/**
+	 * @deprecated 4.0 Method has been replaced by hasOne() and hasOneComponent()
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function has_one($component = null) {
+		if($component) {
+			Deprecation::notice('4.0', 'Please use hasOneComponent() instead');
+			return $this->hasOne($component);
```

Comment:
Returning is nearly always preferable to a massive pyramid of nested conditions. Makes for much simpler to read code.